### PR TITLE
refactor(api,hooks,pagination): agent-search cache helpers, hooks ref unification, and filter comparison (#491)

### PR DESCRIPTION
**Pull request by pawafulu7**
## Summary
- Extract shared cache helpers (`resolveCaches`, `safeReadCache`, `safeWriteCache`) to eliminate duplicate cache instantiation and error handling in agent-search batch/streaming handlers
- Remove dead fallback cache initialization in `createStreamingResponse` and make `caches` parameter required
- Unify `activeRequestId` to ref-only in both `useAgentSearch` and `useArticleQA` hooks, removing redundant useState + useEffect sync that caused unnecessary re-renders
- Replace JSON.stringify order-dependent filter comparison with field-by-field equality in `CursorManager.validateFilters`

## Implementation Details
- **cache-helpers.ts** (new): `resolveCaches(modeContext)` resolves cache instances by mode, `safeReadCache`/`safeWriteCache` wrap cache operations with error logging
- **batch-handler.ts / streaming-handler.ts**: Replaced inline cache instantiation and try-catch blocks with shared helpers
- **useAgentSearch.ts / useArticleQA.ts**: Removed `activeRequestId` state variable and sync useEffect; all stale-request checks now use ref directly
- **cursor-manager.ts**: `validateFilters` now compares filter keys/values individually instead of relying on JSON.stringify key ordering

## Test Results
- cursor-manager.test.ts: 8/8 passed (new - validates key-order independence, edge cases)
- useAgentSearch.contract.test.tsx: 4/4 passed
- useArticleQA.test.ts: 3/3 passed
- TypeScript: 0 errors
- ESLint: 0 errors, 0 warnings
- Build: success

## Checklist
- [x] Tests passing
- [x] No breaking changes
- [x] Cross-review completed (Light tier, Codex)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * レガシーカーソル形式の互換性を追加し、古いカーソルを受け取れるように改善

* **Bug Fixes**
  * カーソルの署名検証と有効期限チェックを強化
  * キャッシュ読み書きの失敗を安全に扱うことでエラー耐性を向上
  * フィルタ比較ロジックを改善し、フィルタ適用の安定性を向上

* **Tests**
  * カーソル検証と符号化/復号の包括的な単体テストを追加

* **Performance**
  * 不要な再レンダリングを削減し応答の安定性を向上

### CODE REVIEW

File: app/api/rag/agent-search/cache-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+): Promise<T | null> {
+  try {
+    return await readFn();
+  } catch (cacheError) {
+    logger.warn(
+      { error: sanitizeError(cacheError), mode },
+      'Cache read failed, treating as miss'
+    );
+    return null;
+  }
```
[the remark this answers is not in the document]
Comment (pawafulu7) [reply app/api/rag/agent-search/cache-helpers.ts]: Intentionally out of scope for this refactoring PR. The miss/error distinction would require expanding the return type across all callers, which is a separate design change.

File: lib/pagination/cursor-manager.ts [hunk 2]
```diff
@@ -48,7 +51,10 @@ export class CursorManager {
   private version: number;
 
   constructor(config: CursorManagerConfig) {
-    this.secret = config.secret || process.env.CURSOR_SECRET || randomBytes(32).toString('hex');
+    this.secret =
+      config.secret ||
+      process.env.CURSOR_SECRET ||
+      randomBytes(32).toString('hex');
```
[the remark this answers is not in the document]
Comment (pawafulu7) [reply lib/pagination/cursor-manager.ts]: This is pre-existing code outside the diff range. Will address in a dedicated cleanup PR if needed.
